da.go

```diff
@@ -1,20 +1,42 @@
 package da
 
+// DA defines very generic interface for interaction with Data Availability layers.
 type DA interface {
+	// Get returns Blob for given ID, or an error.
+	//
+	// Error should be returned if ID is not formatted properly, there is no Blob for given ID or any other client-level
+	// error occurred (dropped connection, timeout, etc).
 	Get(id ID) (Blob, error)
+
+	// Commit creates a Commitment for the given Blob.
 	Commit(blob Blob) (Commitment, error)
+
+	// Submit submits a Blob to Data Availability layer.
+	//
+	// This method is synchronous. Upon successful submission to Data Availability layer, it returns ID identifying blob
+	// in DA and Proof of inclusion.
 	Submit(blob Blob) (ID, Proof, error)
+
+	// Validate validates Commitment against Proof. This should be possible without retrieving Blob.
 	Validate(commit Commitment, proof Proof) (bool, error)
 }
 
+// BatchDA is a batched version of DA interface.
 type BatchDA interface {
```

Comment:
what's the benefit of having both a `BatchDA` and `DA` vs just having the `DA` accept slices? You can still submit a single item to the `BatchDA`, so it feels like unnecessary duplication.

Reply:
agree. just `type DA interface {` that supports array should suffice.

Reply:
The batch operations can be atomic i.e. either all blobs are submitted successfully or none is. This way we can rollback and try again with all pending blobs without having to worry about which ones went through and which ones didn't.

Reply:
Updated to `[]` everywhere.